### PR TITLE
feat: still display replies for deleted comment

### PR DESCRIPTION
**Pull request by benfurber**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Still show the nest comments (replies) when a parent comment is deleted. 

## Git Issues

Closes #

## Screenshots/Videos

https://github.com/ONEARMY/community-platform/assets/16688508/02b19d97-57ce-447b-92eb-963e9f22718b

### CODE REVIEW

**Comment by benfurber** [comment]
@davehakkens Still working on this but wanted to show you the user display I was thinking. 

When a comment (with replies) is marked `_deleted`:
1. The comment and all details are removed.
2. The existing replies are still visible.
3. No new replies can be added.

**Comment by benfurber** [comment]
Happy with this approach @davehakkens?

File: packages/components/src/CommentItem/CommentItem.tsx [hunk 1]
```diff
@@ -83,119 +86,123 @@ export const CommentItem = (props: IProps) => {
...
+        )}
       </Flex>
 
       <Modal width={600} isOpen={showEditModal}>
         <EditComment
           comment={text}
-          handleSubmit={(commentText) => {
-            handleEdit && handleEdit(_id, commentText)
+          handleSubmit={async (commentText) => {
+            handleEdit && (await handleEdit(_id, commentText))
```
Comment (mariojsnunes) [inline packages/components/src/CommentItem/CommentItem.tsx:205]: the `?` can be removed from all the `IProps` functions so here we don't need to check if the function is defined

File: src/stores/Discussions/discussions.store.tsx [hunk 2]
```diff
@@ -297,15 +301,23 @@ export class DiscussionStore extends ModuleStore {
     })
   }
 
+  private _showDiscussion(
```
Comment (mariojsnunes) [inline src/stores/Discussions/discussions.store.tsx:304]: I think this logic doesn't need to be extracted from `_updateDiscussion`. As a reader, I need to jump somewhere else to understand what's going on.

File: src/stores/Discussions/discussions.store.tsx [hunk 3]
```diff
@@ -40,21 +41,18 @@ export class DiscussionStore extends ModuleStore {
       )[0] || null
 
     if (foundDiscussion) {
-      return foundDiscussion
+      return this._showDiscussion(foundDiscussion)
```
Comment (mariojsnunes) [inline src/stores/Discussions/discussions.store.tsx:44]: I think this change isn't needed, due to the if above, `_showDiscussion` would always return the same

**Pull request merged**